### DISCUSSION TITLE
Expose plugin hooks

### DISCUSSION THREAD

**Discussion by mAAdhaTTah** [coddingtonbear/obsidian-local-rest-api#96] [Ideas]
Love the plugin, I've been using it with the Chrome extension quite a bit. I was wondering if you'd given any thought to how other plugins could hook in to provide their own routes. Is there any suggested/blessed way to do that?

**Comment by coddingtonbear** [comment]
I think this is an *excellent* idea, but I haven't done enough research to know what options might be available in Obsidian's plugin architecture for other plugins to advertise their own API endpoints.  Do you happen to know what's available, or are you aware of other plugins that interact with other plugins somehow?  The only example I'm aware of is https://github.com/liamcain/obsidian-daily-notes-interface and I'm vaguely aware that it unfortunately isn't perfectly reliable.

**Reply by mAAdhaTTah** [reply]
The most common example is plugins depending on dataview, which has [docs on developing against it](https://blacksmithgu.github.io/obsidian-dataview/resources/develop-against-dataview/).

**Reply by coddingtonbear** [reply]
I've came across the Dataview docs around this.  Do you happen to have any insight into the actual mechanisms underpinning the API provided, though -- it's one thing to see that developing against a particular library is possible, it's entirely another to understand how that library provides such functionality.

**Reply by mAAdhaTTah** [reply]
The DataView plugin object has an `api` property that constitutes its "public API". The direct mechanism for getting access to that api is [here](https://github.com/blacksmithgu/obsidian-dataview/blob/master/src/index.ts#L53-L56); it takes the `app` and plucks it off appropriately and it falls back to a global variable if you don't provide the app. The actual API is [here](https://github.com/blacksmithgu/obsidian-dataview/blob/master/src/api/plugin-api.ts#L77) and reflects what is documented [here](https://blacksmithgu.github.io/obsidian-dataview/api/code-reference/).

**Comment by coddingtonbear** [comment]
This exists now, by the way, @mAAdhaTTah -- see https://github.com/coddingtonbear/obsidian-local-rest-api/wiki/Adding-your-own-API-Routes-via-an-Extension
